container-images/tcib/base/os/horizontest/run_horizontest.sh

```diff
@@ -128,12 +128,28 @@ function delete_custom_resources {
 }
 
 pushd ${HORIZONTEST_DIR}
+GIT_CLONE_ARGS=()
 if [[ ${REPO_URL} == *redhat.com* ]]; then
-    git -c http.sslVerify=false clone ${REPO_URL} ${HORIZONTEST_DIR}/horizon
-else
-    git clone ${REPO_URL} ${HORIZONTEST_DIR}/horizon
+    GIT_CLONE_ARGS+=(-c http.sslVerify=false)
 fi
+
+_trial=0
+_limit=5
+_delay=30
+
+GIT_CLONE_CMD=(git "${GIT_CLONE_ARGS[@]}" clone "${REPO_URL}" "${HORIZONTEST_DIR}/horizon")
+
+until ("${GIT_CLONE_CMD[@]}"); do
+    if [ "${_trial}" -lt "${_limit}" ]; then
+        (( _trial += 1 ))
+        echo "Git clone failed; retrying in ${_delay} seconds..."
+        sleep "${_delay}"
+    else
+        exit 1
+    fi
+done
 chown -R horizontest:horizontest horizon
+
 pushd horizon
 if [[ ${REPO_URL} == *redhat.com* ]]; then
```

Comment:
this and 4 lines below can be now nicely replaced with
```
git "${GIT_CLONE_ARGS[@]}" pull --rebase
```
(I didnt notice before, maye worth renaming to `GIT_CMD_ARGS` then?)